samples/DataSourceWithProcessingAndExtensions/Sample/CustomSourceParser.cs

```diff
@@ -0,0 +1,222 @@
+using System;
+using System.IO;
+using System.Threading;
+using Microsoft.Performance.SDK;
+using Microsoft.Performance.SDK.Processing;
+using Microsoft.Performance.SDK.Extensibility;
+using Microsoft.Performance.SDK.Extensibility.SourceParsing;
+using Newtonsoft.Json;
+
+namespace DataExtensionsSample
+{
+    /// <summary>
+    /// This class parses a file. In doing so it passes data elements from the file into the runtime for further
+    /// processing by source data cookers. Also during parsing, it fills in the DataSourceInfo property which
+    /// describes this source.
```

Comment:
Can we explain how this is hooked up to the CustomDataProcessor? This is done through reflection, right? If that's the case, it could be confusing to someone reading this wondering how the pieces fit together, since there's no obvious link